pkg/process/procutil/process_windows.go

```diff
@@ -0,0 +1,514 @@
+	p.mapToStat(instance, func(stat *Stats) {
+		stat.CPUTime.System = value.Double
+	})
+}
+
+func (p *probe) mapWorkingSet(instance string, value pdhutil.PdhCounterValue) {
+	p.mapToStat(instance, func(stat *Stats) {
+		stat.MemInfo.RSS = uint64(value.Large)
+	})
+}
+
+func (p *probe) mapPoolPagedBytes(instance string, value pdhutil.PdhCounterValue) {
+	p.mapToStat(instance, func(stat *Stats) {
+		stat.MemInfo.VMS = uint64(value.Large)
+	})
+}
+
+func (p *probe) mapIOReadOpsPerSec(instance string, value pdhutil.PdhCounterValue) {
+	p.mapToStat(instance, func(stat *Stats) {
+		stat.IORateStat.ReadRate = value.Double
+	})
+}
+
+func (p *probe) mapIOWriteOpsPerSec(instance string, value pdhutil.PdhCounterValue) {
+	p.mapToStat(instance, func(stat *Stats) {
+		stat.IORateStat.WriteRate = value.Double
+	})
+}
+
+func (p *probe) mapIOReadBytesPerSec(instance string, value pdhutil.PdhCounterValue) {
+	p.mapToStat(instance, func(stat *Stats) {
+		stat.IORateStat.ReadBytesRate = value.Double
+	})
+}
+
+func (p *probe) mapIOWriteBytesPerSec(instance string, value pdhutil.PdhCounterValue) {
+	p.mapToStat(instance, func(stat *Stats) {
+		stat.IORateStat.WriteBytesRate = value.Double
+	})
+}
+
+func getPIDs() ([]int32, error) {
+	var read uint32
+	var psSize uint32 = 1024
+	const dwordSize uint32 = 4
+
+	for {
+		buf := make([]uint32, psSize)
+		if err := windows.EnumProcesses(buf, &read); err != nil {
+			return nil, err
+		}
+		if uint32(len(buf)) == read {
+			psSize += 1024
+			continue
+		}
+		pids := make([]int32, read)
+		for i := range pids {
+			pids[i] = int32(buf[i])
+		}
+		return pids, nil
+	}
+}
+
+func fillProcessDetails(pid int32, proc *Process) error {
+	procHandle, err := openProcHandle(pid)
+	if err != nil {
+		return err
+	}
+	defer windows.Close(procHandle)
+
+	userName, usererr := getUsernameForProcess(procHandle)
+	if usererr != nil {
+		log.Debugf("Couldn't get process username %v %v", pid, err)
+	}
+	proc.Username = userName
+
+	var imagePath, cmdline string
+	if cmdParams, cmderr := winutil.GetCommandParamsForProcess(procHandle, true); cmderr != nil {
+		log.Debugf("Error retrieving command params %v", cmderr)
+	} else {
+		imagePath = cmdParams.ImagePath
+		cmdline = cmdParams.CmdLine
+	}
+
+	proc.Cmdline = parseCmdLineArgs(cmdline)
+	proc.Exe = imagePath
+
+	var CPU windows.Rusage
+	if err := windows.GetProcessTimes(procHandle, &CPU.CreationTime, &CPU.ExitTime, &CPU.KernelTime, &CPU.UserTime); err != nil {
+		log.Errorf("Could not get process times for %v %v", pid, err)
+		return err
+	}
+
+	ctime := CPU.CreationTime.Nanoseconds() / 1000000
+	proc.Stats.CreateTime = ctime
+	return nil
+}
+
+// TODO: deduplicate
+func openProcHandle(pid int32) (windows.Handle, error) {
+	// 0x1000 is PROCESS_QUERY_LIMITED_INFORMATION, but that constant isn't
+	//        defined in x/sys/windows
+	// 0x10   is PROCESS_VM_READ
+	procHandle, err := windows.OpenProcess(0x1010, false, uint32(pid))
```

Comment:
nit: Can we create a constant for 0x1010 and 0x10?

Reply:
yep, this is duplicated in a couple of places, probably best done in a separate PR, it belongs in the `pkg/util/winutil`.